pkgs/build-support/fetchnextcloudapp/default.nix

```diff
@@ -1,32 +1,27 @@
-{ stdenv, fetchurl, ... }:
-{ name
-, url
-, version
+{ stdenv, fetchzip, applyPatches, ... }:
+{ url
 , sha256
 , patches ? [ ]
+, name ? null
+, version ? null
 }:
-stdenv.mkDerivation {
-  pname = "nc-app-${name}";
-  inherit version patches;
-
-  src = fetchurl {
+if name != null || version != null then throw ''
+  `pkgs.fetchNextcloudApp` has been changed to use `fetchzip`.
+  To update, please
+  * remove `name`/`version`
+  * update the hash
+''
+else applyPatches {
+  inherit patches;
+  src = fetchzip {
     inherit url sha256;
+    postFetch = ''
+      pushd $out &>/dev/null
+      if [ ! -f ./appinfo/info.xml ]; then
+        echo "appinfo/info.xml doesn't exist in $out, aborting!"
+        exit 1
+      fi
```

Comment:
Dropping half the logic from the `installPhase` seems like it might lead to issues. Although the previous logic also seems kind of wonky. E.g. why is just taking the first `'*/appinfo/info.xml'` ok, if multiple are found?

The impression that I'm getting is that currently, one could fetch something where the actual nextcloud app isn't in the root folder and the would work, but this will stop working with these changes.

Reply:
The reason is that some tarballs had the appinfo in their root, while others had it in a folder, and yet others had it in a folder in a folder.

Reply:
> The reason is that some tarballs had the appinfo in their root, while others had it in a folder, and yet others had it in a folder in a folder.

Can you point me to such an example, please?

> E.g. why is just taking the first '*/appinfo/info.xml' ok, if multiple are found?

I'm not sure if that's even possible. The previous behavior of just installing a single app even if multiple are in the tarball seems unexpected anyways. I guess if we allow `*/appinfo/info.xml` to exist, we should error out instead IMHO.

> Dropping half the logic from the installPhase seems like it might lead to issues

Just to make sure I'm not missing something, except from the appinfo/info.xml location we're currently discussing, anything else you're missing?

Reply:
> anything else you're missing?

Probably not. I'm not even sure that feature is even relevant/correct, as you said. Just that we probably shouldn't drop it without making sure that it's ok to do so first and maybe also documenting the changed behavior.

Reply:
ping @beardhatcode would be cool to get an example to sort the special-case (and the behavior of the new implementation out) before merging :)

Reply:
I just veriefied with all the apps I ever installed with nix and they now all have `{appname}/appinfo/info.xml`. And thanks to the `stripRoot` of `fetchZip` (which I did not know about) this works :+1: